### PR TITLE
:app + :core — "Why this outfit?" rationale dialog on outfit cards

### PR DESCRIPTION
**Pull request by mikelward**
## Summary

Tap an outfit card on Today to see *why* each garment was picked. Each slot lists its deciding fact(s) — the feels-like min/max, the hour it occurred, and the threshold it crossed — e.g. *"Feels-like low 13°C at 07:00, under 18°C"*. Honours the user's `TemperatureUnit` (°C/°F).

Built from the user-facing prompt: *"too warm for a jacket — show me 5°C at 6pm or let me adjust the threshold."* This PR is the read-only "show me the evidence" half. A stacked follow-up adds the `−1° / +1°` tuning controls (TODO marker on `OutfitRationaleDialog`).

### Domain (`:core:domain`)
- Extracted the previously-hardcoded numbers in `OutfitSuggestion.fromForecast` into a `Thresholds` data class with `DEFAULT`. Same numbers, but now addressable — the follow-up swaps `DEFAULT` for a prefs-backed instance without touching call sites.
- New `OutfitSuggestion.explainFromForecast(forecast, thresholds): OutfitRationale` returning `GarmentReason` per slot. Bottom slot can carry 1–2 `Fact`s because shorts is a two-condition rule (warm max **and** not-too-cold min).
- Added optional `outfitRationale` / `nextOutfitRationale` to `Insight` (nullable, default `null` — old caches deserialize fine, next worker run repopulates).

### App (`:app`)
- `OutfitPreviewCard` is now `Modifier.clickable`, with a small "Why?" hint, opening an `AlertDialog`.
- `InsightCache`: additive `OutfitRationaleDto` round-trip; **no cache key bump** because the new fields are optional.
- New strings, plus a Roborazzi snapshot (`outfit_rationale_dialog`).

### Privacy

Local-only. The new field is **not** consumed by `InsightFormatter`, `ClothesPhraser`, the TTS path, the Gemini insight call, or any notifier — only by the dialog. Nothing new crosses the device boundary.

### Validation gap

Couldn't run `:core:domain:test` or `:app:testDebugUnitTest` locally — the agent sandbox can't reach Google Maven so AGP plugin resolution fails at the root build script. Static-checked the diff; relying on CI for the actual compile + test run.

## Test plan

- [ ] CI: `JVM unit tests` job green (new `OutfitSuggestionTest` + `InsightCache` rationale round-trip)
- [ ] CI: `Android debug build` job green
- [ ] Roborazzi `outfit_rationale_dialog.png` looks right
- [ ] Manual: tap each outfit card, dialog renders feels-like values + times against the user's unit
- [ ] Manual: legacy cache (no rationale) still opens — dialog shows the "no rationale available" fallback

https://claude.ai/code/session_011VmF6s7pAyvVaU9c5SFWsU

---
_Generated by [Claude Code](https://claude.ai/code/session_011VmF6s7pAyvVaU9c5SFWsU)_

### CODE REVIEW

File: app/src/main/kotlin/app/clothescast/ui/today/TodayScreen.kt [hunk 1]
```diff
@@ -450,10 +467,132 @@ internal fun OutfitPreviewCard(
...
+            )
+        }
+    }
+}
+
+@Composable
+private fun formatFact(fact: Fact, unit: TemperatureUnit): String {
+    val symbol = unit.symbol()
+    val observed = fact.observedC.toUnit(unit).roundToInt()
+    val threshold = fact.thresholdC.toUnit(unit).roundToInt()
```
[the remark this answers is not in the document]
Comment (mikelward) [reply app/src/main/kotlin/app/clothescast/ui/today/TodayScreen.kt]: Good catch — fixed in the latest force-push. `formatFact` now drops to one-decimal precision when integer rounding would make the observed and threshold values look equal but they're actually different (the exact "17.6 < 18.0 rendering as '18°C, under 18°C'" case). Common case is still bare integers; the fallback only triggers on the boundary edge.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: core/domain/src/main/kotlin/app/clothescast/core/domain/model/OutfitSuggestion.kt [hunk 2]
```diff
@@ -19,22 +21,148 @@ data class OutfitSuggestion(
...
+            // failed (or both, if both did) so the user sees the actual blocker.
+            val maxOk = forecast.feelsLikeMaxC > thresholds.shortsMinFeelsLikeMaxC
+            val minOk = forecast.feelsLikeMinC > thresholds.shortsMinFeelsLikeMinC
+            val maxFact = Fact(
+                metric = Fact.Metric.FEELS_LIKE_MAX,
+                observedC = forecast.feelsLikeMaxC,
+                observedAt = warmestHour?.time,
+                thresholdC = thresholds.shortsMinFeelsLikeMaxC,
+                comparison = if (maxOk) Fact.Comparison.AT_OR_ABOVE else Fact.Comparison.BELOW,
+            )
```
[the remark this answers is not in the document]
Comment (mikelward) [reply core/domain/src/main/kotlin/app/clothescast/core/domain/model/OutfitSuggestion.kt]: Aligned in the latest force-push: changed both `fromForecast` and `explainFromForecast` to use `>=` so equality lands on `AT_OR_ABOVE` for the shorts cutoffs, matching the new "at least %d°C" wording. Added a boundary-equality test case (`Triple(15.0, 22.0, …, SHORTS)`) to keep the rule operator and the comparison label in lockstep.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -24,6 +24,22 @@
...
+    <!-- "Feels-like low 13°C at 6am, under 18°C" -->
+    <string name="today_rationale_min_below_with_time">Feels-like low %1$d%2$s at %3$s, under %4$d%2$s</string>
+    <!-- "Feels-like low 13°C, under 18°C" — when hourly data is missing -->
+    <string name="today_rationale_min_below">Feels-like low %1$d%2$s, under %3$d%2$s</string>
+    <string name="today_rationale_min_above_with_time">Feels-like low %1$d%2$s at %3$s, above %4$d%2$s</string>
+    <string name="today_rationale_min_above">Feels-like low %1$d%2$s, above %3$d%2$s</string>
+    <string name="today_rationale_max_below_with_time">Feels-like high only %1$d%2$s at %3$s, under %4$d%2$s</string>
+    <string name="today_rationale_max_below">Feels-like high only %1$d%2$s, under %3$d%2$s</string>
+    <string name="today_rationale_max_above_with_time">Feels-like high %1$d%2$s at %3$s, above %4$d%2$s</string>
+    <string name="today_rationale_max_above">Feels-like high %1$d%2$s, above %3$d%2$s</string>
```
[the remark this answers is not in the document]
Comment (mikelward) [reply app/src/main/res/values/strings.xml]: Applied the suggested wording — "above" → "at least" on every AT_OR_ABOVE prose variant. Also switched the format args from `%d` to `%s` so the same templates can carry a one-decimal fallback when integer rounding would make the observed and threshold values indistinguishable (paired with the codex P2 fix above).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/src/main/kotlin/app/clothescast/ui/today/TodayScreen.kt [hunk 4]
```diff
@@ -414,8 +426,13 @@ internal fun OutfitPreviewCard(
     outfit: OutfitSuggestion,
     label: String,
     modifier: Modifier = Modifier,
+    rationale: OutfitRationale? = null,
+    temperatureUnit: TemperatureUnit = TemperatureUnit.CELSIUS,
 ) {
-    Card(modifier = modifier) {
+    var showRationale by remember { mutableStateOf(false) }
+    Card(
+        modifier = modifier.clickable { showRationale = true },
```
[the remark this answers is not in the document]
Comment (mikelward) [reply app/src/main/kotlin/app/clothescast/ui/today/TodayScreen.kt]: Applied — the card now uses Material3's `Card(onClick = …)` overload with `@OptIn(ExperimentalMaterial3Api::class)` on the composable. Matches `SettingsNavRow`'s pattern.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: core/domain/src/test/kotlin/app/clothescast/core/domain/model/OutfitSuggestionTest.kt [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        // Pinned cases that exercise each branch — keeps fromForecast and
+        // explainFromForecast from drifting.
+        val cases = listOf(
+            Triple(7.5, 9.0, OutfitSuggestion(OutfitSuggestion.Top.THICK_JACKET, OutfitSuggestion.Bottom.LONG_PANTS)),
+            Triple(13.0, 17.0, OutfitSuggestion(OutfitSuggestion.Top.SWEATER, OutfitSuggestion.Bottom.LONG_PANTS)),
+            Triple(16.0, 26.0, OutfitSuggestion(OutfitSuggestion.Top.TSHIRT, OutfitSuggestion.Bottom.SHORTS)),
+        )
+        cases.forEach { (min, max, expected) ->
+            OutfitSuggestion.fromForecast(forecast(min, max)) shouldBe expected
+        }
```
[the remark this answers is not in the document]
Comment (mikelward) [reply core/domain/src/test/kotlin/app/clothescast/core/domain/model/OutfitSuggestionTest.kt:159]: Added pinned equality cases to `rationale matches the suggestion across boundary cases` — exactly 8°C (jacket/sweater), 18°C (sweater/t-shirt), and the 15/22°C shorts boundary. The shorts case caught a real bug while I was at it: the rule used strict `>` but the comparison label was `AT_OR_ABOVE`, which the equality test now keeps honest (rule + label are both inclusive in the latest force-push).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: core/domain/src/main/kotlin/app/clothescast/core/domain/model/OutfitSuggestion.kt [hunk 6]
```diff
@@ -19,22 +21,148 @@ data class OutfitSuggestion(
...
+    /**
+     * The thresholds [fromForecast] uses to map a forecast onto a top + bottom. Pulled out
+     * as a named object so both the suggestion and the rationale read the same numbers,
+     * and so a future PR can swap [DEFAULT] for a user-tunable instance plumbed through
+     * preferences without touching the call sites in [GenerateDailyInsight].
+     */
+    data class Thresholds(
+        val sweaterMaxFeelsLikeMinC: Double,
+        val tshirtMinFeelsLikeMinC: Double,
+        val shortsMinFeelsLikeMaxC: Double,
```
[the remark this answers is not in the document]
Comment (mikelward) [reply core/domain/src/main/kotlin/app/clothescast/core/domain/model/OutfitSuggestion.kt:33]: Fair — the name reads like sweater's max, not jacket's. Leaving as-is for now though: in the stacked follow-up (#238) these field names become user-visible `Fact.ThresholdKind` enum values + DataStore preference keys (`outfit_threshold_sweater_max_feels_like_min_c` etc.), and the field name doubles as the kind label that the rationale dialog's `−1°` / `+1°` controls bind to. Renaming once both PRs are stacked has a non-trivial blast radius (enum, prefs keys, cache schema, tests) for a doc-level concern. Will revisit if a clearer name surfaces — `thickJacketBelowFeelsLikeMinC` is a candidate but readers find it equally tricky to parse, since the field's value (8.0) is a transition point between *two* garments depending on which side of the comparison you read it from.

---
_Generated by [Claude Code](https://claude.ai/code)_
